pyro/util.py

```diff
@@ -181,6 +182,26 @@ def ng_zeros(*args, **kwargs):
     return Variable(p_tensor if retype is None else p_tensor.type_as(retype), requires_grad=False)
 
 
+def is_nan(x):
+    """
+    A convenient function to check if a Tensor contains all nan; also works with numbers
+    and torch.autograd.Variable
+    """
+    if isinstance(x, numbers.Number):
+        return x != x
+    return all(x != x)
```

Comment:
I believe this should use PyTorch via
```py
return (x !=  x).all()
```

Reply:
@fritzo It seems that they are the same in my system. Anyway, I made the changes based on my your suggestions.

Reply:
```py
>>> all(torch.ones(2))
True
>>> all(torch.ones(2, 2))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/fritzobermeyer/pytorch-dev/2.7/torch/tensor.py", line 168, in __bool__
    " containing more than one value is ambiguous")
RuntimeError: bool value of torch.FloatTensor containing more than one value is ambiguous
```